### ISSUE TITLE
Support for remote branches, too? (or why not?)

### ISSUE THREAD

**Issue by dwiyatci** [m-hall/vscode-git-branch-sidebar#8]
Hi, Mia! :) I'm recently moving from WebStorm to VSCode and found this extension super-helpful and so simple-to-use (deleting in one-click and pulling in two-click in a flat list are my favs 🤩). Thanks a lot for the efforts! 🙏

I wonder if the **remote branches** could be listed as well in addition to the local ones, and whether it makes sense to add such a functionality (if not, I just wanted to know the design consideration/assumption taken back then).

Actually, I also have two other queries but I created separate issue for them: #9 and #10.

Thanks again! :))

**Comment by m-hall** [comment]
In my case, the reason that I didn't add remote branches originally was because I wanted it to manage my own local branches, as I'm often working on several issues at one time, and each in their own branch.
I found VSCode's "select branch" popup a bit hard to use when dealing with local branch, but sufficient for dealing with remote branches, which I do much more rarely.

I have considered adding a separate tree for remote branches, but with the ability to have multiple repos in a single workspace, I decided that would be too much of a hassle to manage, plus it would need to keep requesting updates from the remote more frequently that I intended when I created it. Though the last couple features added have necessitated that anyway.

Generally if you a want more "complete" interface for working with Git, I would recommend using something like GitLens. I just found it too cumbersome for my needs 95% of the time, so I usually stick with this little extension for most things, then switch to GitLens for more complex things.

To be clear: I'm not opposed to it, but it wasn't my intention when I built this extension, as I specifically want to keep it tiny, and not on the scale of something much more feature complete.

**Comment by dwiyatci** [comment]
Thanks heaps for the extended explanation and clarification!^^ Yah, coming from WebStorm I guess I'm still used to the UI where you have the list of local and remote branches, and you can do an operation to any branch directly from the list without switching/checking-out the branch first (your extension resembles this interaction), unlike the VSCode's built-in Git status bar.

<img width="880" alt="Screenshot 2024-01-12 at 14 34 42" src="https://github.com/m-hall/vscode-git-branch-sidebar/assets/4405796/4136288e-4f6e-4ce2-b3ca-cbad0928d225">

---
I know GitLens exists, and I tried it out. But I don't quite like it for two reasons:
1. It's too bloated. It has bunch of features that I don't need ("You wanted a banana but what you got was a gorilla holding the banana and the entire jungle" 🍌🦍😅) and it clutters all over-- against my spirit of minimalism :v (https://pointersgonewild.com/2018/02/18/minimalism-in-programming/)
2. It's kinda doing a funny business pushing us to use the paid version of the extension, and I don't wanna deal with GitKraken license atm.

Anyway, thanks a lot again for enhancing the extension by closing #9 and #10. I'm pretty happy now to use your extension along with the built-in VSCode Git + the other VSCode's Git extensions: Git Graph + Git Blame + Git Prefix – I believe it should cover all my Git use cases for now ♡

Have a nice weekend!! 🥳

**Comment by m-hall** [comment]
Yeah I have similar feelings regarding GitLens, I use it very sparingly, mostly for viewing the commit history. It was actually the main reason that I chose to build this extension.

I'm going to leave this ticket open for now, as I do think it would be a good idea, but I want to make sure I have a good way for it to work before I tackle it.
